### PR TITLE
feat(difc): structured logging for every filtered item

### PR DESCRIPTION
**Pull request by lpcox**
## Summary

Adds per-item structured logging when DIFC filtering removes objects from tool responses. This enables post-processing reports that show exactly what was filtered, why, and provide enough context to link back to the GitHub object.

## Changes

- **`FilteredItemDetail` struct** (`resource.go`): Pairs each filtered `LabeledItem` with the denial reason from the evaluator
- **`FilterCollection`** (`evaluator.go`): Now captures `EvaluationResult.Reason` per denied item
- **`difc_log.go`** (new): `logFilteredItems()` extracts metadata from raw JSON data and emits structured JSON log entries via the file logger
- **Phase 5 integration** (`unified.go`): Calls `logFilteredItems()` whenever items are filtered
- **`difc_log_test.go`** (new): Unit tests verifying the structured log output format, JSON validity, expected fields, and per-server/unified log routing

## Log Entry Format

Each filtered item produces a JSON entry logged at `[DIFC-FILTERED]` prefix:

```json
{
  "server_id": "github",
  "tool_name": "list_issues",
  "description": "issue:repo#42",
  "reason": "integrity too low",
  "secrecy_tags": ["private:org/repo"],
  "integrity_tags": ["none"],
  "author_association": "NONE",
  "author_login": "user",
  "html_url": "https://github.com/...",
  "number": "42"
}
```

Entries are written to both per-server log files and the unified `mcp-gateway.log` for easy grepping.

## Testing

- All existing tests updated for the `FilteredItemDetail` type change
- New tests in `difc_log_test.go` assert that `logFilteredItems` emits valid JSON with the `[DIFC-FILTERED]` prefix, writes to both per-server and unified logs, and handles edge cases (nil labels, non-map data, empty filtered list, multiple items)
- `make agent-finished` passes (format, build, lint, unit + integration tests)

---

📍 Connect Copilot coding agent with [Jira](https://gh.io/cca-jira-docs), [Azure Boards](https://gh.io/cca-azure-boards-docs) or [Linear](https://gh.io/cca-linear-docs) to delegate work to Copilot in one click without leaving your project management tool.

### CODE REVIEW

File: internal/server/difc_log.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		if err != nil {
+			logger.LogWarnWithServer(serverID, "difc",
+				"[DIFC-FILTERED] %s | %s | description=%s | reason=%s (json marshal failed: %v)",
+				serverID, toolName, entry.Description, entry.Reason, err)
+			continue
+		}
+
+		logger.LogInfoWithServer(serverID, "difc",
+			"[DIFC-FILTERED] %s", string(entryJSON))
+	}
```
[the remark this answers is not in the document]
Comment (lpcox) [reply internal/server/difc_log.go:45]: @copilot apply changes based on [this feedback](https://github.com/github/gh-aw-mcpg/pull/2066#discussion_r2949757459)
